pkg/crypto/crypto_test.go

```diff
@@ -0,0 +1,358 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"github.com/mr-tron/base58/base58"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func testKeccak(t *testing.T, d, h string) {
+	data, err := hex.DecodeString(d)
+	if assert.NoError(t, err) {
+		actual := Keccak256(data)
+		expected, err := hex.DecodeString(h)
+		if assert.NoError(t, err) {
+			assert.Equal(t, expected, actual[:])
+		}
+	}
+}
+
+func TestKeccak1(t *testing.T) {
+	const (
+		dataString = "0100000000000000000000000000000000000000000000000000000000000000"
```

Comment:
We should move all tests and benchmarks to slices of testcases https://blog.golang.org/subtests